drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -377,116 +368,60 @@ local function migrate_light(driver, device, parent_device_id)
-
-            for _, light in ipairs(light_resource.data or {}) do
-              local profile_ref
-
-              if light.color then
-                profile_ref =
-                "white-and-color-ambiance" -- all color light products support `white` (dimming) and `ambiance` (color temp)
-              elseif light.color_temperature then
-                profile_ref = "white-ambiance" -- all color temp products support `white` (dimming)
-              elseif light.dimming then
-                profile_ref = "white"    -- `white` refers to dimmable and includes filament bulbs
-              else
-                log.warn_with({ hub_logs = true },
-                  string.format(
-                    "Light resource [%s] does not seem to be A White/White-Ambiance/White-Color-Ambiance device, currently unsupported"
-                    ,
-                    svc_info.rid
-                  )
-                )
-                return
-              end
-              local updated_parent_id = parent_device_id or known_dni_to_device_map[bridge_id].id
-              local new_metadata = {
-                profile = profile_ref,
-                manufacturer = device_data.product_data.manufacturer_name,
-                model = device_data.product_data.model_id,
-                vendor_provided_label = device_data.product_data.product_name,
-              }
-              device:try_update_metadata(new_metadata)
-
-              Discovery.light_state_disco_cache[light.id] = {
-                on = light.on,
-                color = light.color,
-                dimming = light.dimming,
-                color_temp = light.color_temperature,
-                mode = light.mode,
-                parent_device_id = updated_parent_id,
-                hue_device_id = light.owner.rid
-              }
-              light_migrated = true
-              logged_light_added(driver, device, updated_parent_id, resource_id)
-            end
-          end, "[migrate_light]"
-        )
-        if light_migrated then return end
-        time_remaining = math.max(0, timeout_time - cosock.socket.gettime())
-      end
-      if not light_migrated then
-        log.warn_with({ hub_logs = true }, string.format(
-          "Failed to migrate bulb %s, marking \"stray\".", (device.label or device.id or "unknown device")
-        ))
-        driver.stray_bulb_tx:send({
-          type = StrayDeviceMessageTypes.NewStrayLight,
-          driver = driver,
-          device = device
-        })
-      end
-    end,
-    "migrate light re-scan"
-  )
+  local new_metadata = {
+    profile = profile_ref,
+    manufacturer = light_resource.hue_device_data.product_data.manufacturer_name,
+    model = light_resource.hue_device_data.product_data.model_id,
+    vendor_provided_label = light_resource.hue_device_data.product_data.product_name,
+  }
+  device:try_update_metadata(new_metadata)
+
+  local updated_parent_id = parent_device_id or known_dni_to_device_map[bridge_id].id
```

Comment:
You already have `bridge_device` from line 332 and validate that it exists. you could just use `bridge_device.id` instead of `updated_parent_id`

Reply:
Good callout. This is a leftover from a time when the `parent_device_id` was accidentally getting set to the Hub's ID instead of the Hue Bridge during migration. That bug was fixed a long time ago and I wouldn't expect devices to be experiencing this anymore.